packages/flutter_tools/lib/src/commands/devices.dart

```diff
@@ -32,20 +33,13 @@ class DevicesCommand extends FlutterCommand {
   @override
   final String description = 'List all connected devices.';
 
-  Duration get timeout {
-    if (argResults['timeout'] == null) {
-      return null;
-    }
-    if (_timeout == null) {
-      final int timeoutSeconds = int.tryParse(stringArg('timeout'));
-      if (timeoutSeconds == null) {
-        throwToolExit( 'Could not parse -t/--timeout argument. It must be an integer.');
-      }
-      _timeout = Duration(seconds: timeoutSeconds);
+  @override
+  Future<void> validateCommand() {
+    if (argResults['timeout'] != null) {
+      throwToolExit('--timeout has been deprecated, use --${FlutterOptions.kDeviceTimeout} instead');
```

Comment:
@zanderso What do you think of this hard deprecation?  I doubt this is commonly used.  I didn't want to silently drop it.  I could make it more complicated and print this warning, but then try to pass the deprecated `timeout` down into `deviceDiscoveryTimeout`.

Reply:
Plumbing it to `deviceDiscovertyTimeout` sounds like the right way to deprecate this. It can then be entirely removed after the stable branch cut.